src/MetalPerformanceShaders/MPSDefs.cs

```diff
@@ -99,8 +111,112 @@ public struct MPSImageHistogramInfo {
 		public Vector4 MaxPixelValue;
 	}
 
+	public enum MPSMatrixDecompositionStatus
+	{
+		Success = 0,
+		Failure = -1,
+		Singular = -2,
+		NonPositiveDefinite = -3,
+	}
+
 	// MPSTypes.h
 	// FIXME: public delegate IMTLTexture MPSCopyAllocator (MPSKernel filter, IMTLCommandBuffer commandBuffer, IMTLTexture sourceTexture);
 	public delegate NSObject MPSCopyAllocator (MPSKernel filter, NSObject commandBuffer, NSObject sourceTexture);
 	// https://trello.com/c/GqtNId1C/517-generator-our-block-delegates-needs-to-use-wrapper-for-protocols
-}
\ No newline at end of file
+
+	[TV (11, 0), Mac (10, 13, onlyOn64: true), iOS (11, 0)]
+	[Native]
+	public enum MPSRnnSequenceDirection : nuint {
+		Forward = 0,
+		Backward,
+	}
+
+	[TV (11, 0), Mac (10, 13, onlyOn64: true), iOS (11, 0)]
+	[Native]
+	public enum MPSRnnBidirectionalCombineMode : nuint {
+		None = 0,
+		Add,
+		Concatenate,
+	}
+
+	[TV (11, 0), Mac (10, 13, onlyOn64: true), iOS (11, 0)]
+	public enum MPSCnnNeuronType {
+		None = 0,
+		ReLU,
+		Linear,
+		Sigmoid,
+		HardSigmoid,
+		TanH,
+		Absolute,
+		SoftPlus,
+		SoftSign,
+		Elu,
+		PReLU,
+		ReLUN,
+		Count,
+	}
+
+	[TV (11, 0), Mac (10, 13, onlyOn64: true), iOS (11, 0)]
+	[Native]
+	public enum MPSCnnBinaryConvolutionFlags : nuint {
+		None = 0,
+		UseBetaScaling = 1 << 0,
+	}
+
+	[TV (11, 0), Mac (10, 13, onlyOn64: true), iOS (11, 0)]
+	[Native]
+	public enum MPSCnnBinaryConvolutionType : nuint {
+		BinaryWeights = 0,
+		Xnor,
+		And,
+	}
+
+	[TV (11, 0), Mac (10, 13, onlyOn64: true), iOS (11, 0)]
+	[Native]
+	public enum MPSNnPaddingMethod : nuint {
+		AlignCentered = 0,
+		AlignTopLeft = 1,
+		AlignBottomRight = 2,
+		AlignReserved = 3,
+		AddRemainderToTopLeft = 0 << 2,
+		AddRemainderToTopRight = 1 << 2,
+		AddRemainderToBottomLeft = 2 << 2,
+		AddRemainderToBottomRight = 3 << 2,
+		SizeValidOnly = 0,
+		SizeSame = 1 << 4,
+		SizeFull = 2 << 4,
+		SizeReserved = 3 << 4,
+		Custom = (1 << 14),
+		SizeMask = 2032,
+		ExcludeEdges = (1 << 15),
+	}
+
+	[TV (11, 0), Mac (10, 13, onlyOn64: true), iOS (11, 0)]
+	[Native]
+	public enum MPSDataLayout : nuint
+	{
```

Comment:
Minor, move to line above, at least the structs/enums above and below are that way :)